### DISCUSSION TITLE
How are files organized into packages?

### DISCUSSION THREAD

**Discussion by chenzhekl** [odin-lang/Odin#1147] [Q&A] [answered]
Hi,

Thank you for the great language. I am learning Odin and get confused about the package organization. It seems quite close to Go but I cannot be sure about that. I tried looking for the answer in the documentation but got nothing. Could anyone please explain it to me?

Thanks

**Comment by Tetralux** [comment, marked as answer]
This is a bit late, but basically: packages are just folders with `.odin` files in them.
You can import them with `import "path/to/folder"`, with the path relative to the current file.
You can also import packages from a collection: `import "stuff:path/to/folder"` if you pass `-collection:stuff=path/to/root/folder`.

**Reply by Inc0n** [reply]
Hi, if you don't mind me asking some further questions:

So individual files from a package cannot be imported, but the whole package will? Is that what it implies.

**Reply by Kelimion** [reply]
That is correct, all* files from the package will be imported.

But only the procedures that are used within that package will be code-genned and linked, so you're not paying for unused parts of the package.

You can also make a package with a single file in it and import that, which is as simple as making a directory and putting a .odin file in it.

* There are the `*_windows.odin`, `*_linux.odin` suffixes so those are only imported for those build targets, and there are `// build` comments at the top of packages, but by and large: Yes, all files within a package are imported.
